content/en/docs/setup/install/virtual-machine/index.md

```diff
@@ -182,45 +174,121 @@ Run the following commands on the virtual machine you want to add to the Istio m
+1. Install `cluster.env` within the directory `/var/lib/istio/envoy/`.
 
     {{< text bash >}}
-    $ sudo mkdir -p /etc/certs
-    $ sudo cp {root-cert.pem,cert-chain.pem,key.pem} /etc/certs
+    $ sudo cp "${HOME}"/cluster.env /var/lib/istio/envoy/cluster.env
     {{< /text >}}
 
-1. Install `cluster.env` within `/var/lib/istio/envoy/`.
+1. Install `sidecar.env` within the directory`/var/lib/istio/envoy/`.
 
     {{< text bash >}}
-    $ sudo cp cluster.env /var/lib/istio/envoy
+    $ sudo cp "${HOME}"/sidecar.env /var/lib/istio/envoy/sidecar.env
     {{< /text >}}
 
 1. Add the istiod host to `/etc/hosts`.
 
     {{< text bash >}}
-    $ sudo sh -c 'cat hosts-addendum >> /etc/hosts'
+    $ sudo sh -c 'cat $(eval echo ~$SUDO_USER)/hosts-addendum >> /etc/hosts'
+    {{< /text >}}
+
+1. Install the root certificate in the directory  `/etc/certs`
+
+    {{< text bash >}}
+    $ sudo cp "${HOME}"/root-cert.pem /var/run/secrets/istio/root-cert.pem
     {{< /text >}}
 
 1. Transfer ownership of the files in `/etc/certs/` and `/var/lib/istio/envoy/` to the Istio proxy.
 
     {{< text bash >}}
-    $ sudo chown -R istio-proxy /etc/certs /var/lib/istio/envoy
+    $ sudo mkdir -p /etc/istio/proxy
+    $ sudo chown -R istio-proxy /var/lib/istio /etc/certs /etc/istio/proxy  /var/run/secrets
     {{< /text >}}
 
-1. Start Istio within the virtual machine.
+## Setup policies for virtual machine
+1. Get the external ip or internal ip of the virtual machine and store to `"${VM_IP}"`
 
     {{< text bash >}}
-    $ sudo systemctl start istio
+    $ VM_IP="<the external ip or internal ip of the virtual machine>"
+    {{< /text >}}
+    
+1. Create a new service
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl -n "${SERVICE_NAMESPACE}" apply -f -
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: cloud-vm
+      labels:
+        app: cloud-vm
+    spec:
+      ports:
+      - port: 7070
+        name: grpc
+      - port: 8090
+        name: http
+      selector:
+        app: cloud-vm
+    EOF
+    {{< /text >}}
+1. Create workload entry
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl -n "${SERVICE_NAMESPACE}" apply -f -
+      apiVersion: networking.istio.io/v1beta1
+      kind: WorkloadEntry
+      metadata:
+        name: "${VM_NAME}"
+        namespace: "${SERVICE_NAMESPACE}"
+      spec:
+        address: "${VM_IP}"
+        labels:
+          app: gce-vm
+        serviceAccount: "${SERVICE_ACCOUNT}"
+      EOF
     {{< /text >}}
 
+## Start Istio within the virtual machine.
+
+1. start the istio agent
+
+    {{< text bash >}}
+    $ sudo systemctl start istio
+    {{< /text >}}
+    
 ## Uninstall
+Stop the istio
+    {{< text bash >}}
+    $ sudo systemctl stop istio
+    {{< /text >}}
+
+Remove existing istio-sidecar package
+    {{< text bash >}}
+    $ sudo dpkg -r istio-sidecar
+    $ dpkg -s istio-sidecar
+    {{< /text >}}
 
 To uninstall Istio, run the following command:
```

Comment:
Do you plan to add instructions to show vm and pod communication? Will it be in this doc or other docs?

Reply:
it will be another doc this is only for installation. It is shown on https://preliminary.istio.io/latest/docs/examples/virtual-machines/single-network/